synapse/util/ratelimitutils.py

```diff
@@ -51,6 +52,34 @@ def new_limiter() -> "_PerHostRatelimiter":
             str, "_PerHostRatelimiter"
         ] = collections.defaultdict(new_limiter)
 
+        # We track the number of affected hosts per time-period so we can
+        # differentiate one really noisy homeserver from a general
+        # ratelimit tuning problem across the federation.
```

Comment:
I feel like I may be going at this backwards. Is there a better way to track the number of distinct/unique things in a set for a given period of time?

We can't just tack on the `host` to the labels of the existing counters because it is unbounded and could blow up our cardinality.

Reply:
I think having a gauge of "number of hosts that are currently being ratelimited" makes sense.

Reply:
I can't think of a better way of doing it really.

Reply:
whew, glad this makes sense 😅